test/common/router/shadow_writer_impl_test.cc

```diff
@@ -59,5 +59,48 @@ TEST(ShadowWriterImplTest, All) {
   callback->onFailure(Http::AsyncClient::FailureReason::Reset);
 }
 
+TEST(ShadowWriterImplTest, AllWithHostAndPort) {
```

Comment:
Since most of this is duplicated between the first testa case and this one with the port, can you refactor it to have the host values be passed in as a parameter?

Reply:
Yes, thanks for suggesting this.